homeassistant/components/vulcan/__init__.py

```diff
@@ -46,5 +46,8 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
         hass.data[DOMAIN].pop(entry.entry_id)
+        hass.data[DOMAIN]["students_number"] = len(
+            hass.config_entries.async_entries(DOMAIN)
```

Comment:
This should probably only include the loaded config entries. So we should probably check the config entry state when enumerating here. We also need to remove one count since the current entry is still loaded at this point but will not be loaded when we return here.

Reply:
Do you mean something like `hass.data[DOMAIN]["students_number"] = len([entry for entry in hass.config_entries.async_entries(DOMAIN) if entry.state == ConfigEntryState.LOADED]) - 1`?

Reply:
Looking at where we use this, I would rather consider removing this data point and always set the entity and device name including the student name. Why do we have this case?

Reply:
Sure, rn it's there to simplify name when there is only one student added but i'ts not necessary